tests/ui/redundant_pattern_matching_result.rs

```diff
@@ -125,3 +126,30 @@ const fn issue6067() {
         Err(_) => true,
     };
 }
+
+fn issue10726() {
+    match Ok::<i32, i32>(42) {
+        Ok(_) => true,
+        _ => false,
+    };
+
+    match Ok::<i32, i32>(42) {
+        Err(_) => true,
+        _ => false,
+    };
+
+    match Err::<i32, i32>(42) {
+        Ok(_) => true,
+        _ => false,
+    };
```

Comment:
These lints only check the patterns and branches, the scrutinee (Expression used by the `match` to match) is ignored by the lint. The first and last match in the marked code are equivalent. Could you instead use the following tests:

```
    // This is optional, but it makes the examples easier
    let x: Result<i32, i32> = Ok(42)

    match x {
        Ok(_) => true,
        _ => false,
    };

    match x {
        Ok(_) => false,
        _ => true,
    };

    match x {
        Err(_) => true,
        _ => false,
    };

    match x {
        Err(_) => false,
        _ => true,
    };

    // Don't lint
    match x {
        Err(16) => false,
        _ => true,
    };

    // Don't lint
    match x {
        Ok(16) => false,
        _ => true,
    };
```

And also adjust these for the `Option` tests